core/src/main/scala/filodb.core/store/InMemoryColumnStore.scala

```diff
@@ -165,6 +165,25 @@ trait InMemoryColumnStoreScanner extends ColumnStoreScanner {
       toIterator
   }
 
+  def multiPartScan(projection: RichProjection, version: Int, partitions: Seq[Any]):
+    Iterator[(BinaryPartition, RowMapTreeLike)] = {
+    partitions.map { partition => {
```

Comment:
I think it's better to do `partitions.toIterator` rather than do a map and then convert everything to an Iterator.  That way, the chunkRowMap lookup is lazy, in case there are a lot of partitions this will save memory.